### PR TITLE
Enhance README.md and list features

### PR DESCRIPTION
**Pull request by torkleyy**
@kvark pointed out that it would be nice the see the overall state of Amethyst. Obviously, we'll want to put shiny screenshots here in the future, but I think this is a good start.

I also deleted the reference to Bitsquid since it is both discontinued and I don't think we're still taking inspirations from it (I barely know it). Some edits have been made to the `CONTRIBUTING.md`, mainly deleting outdated stuff.

[Rendered version](https://github.com/torkleyy/amethyst/blob/93abf7971b00e241518a563bc97a4dec8619d405/README.md)

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/704)

### CODE REVIEW

**Comment by kvark** [comment]
Thank you for addressing this promptly!

I think the current design has too many bullet points :) as in - it's great that Amethyst has features, but it's not the most welcoming README to read through them. I'd go as far as say that features could be hidden behind a click, say, in a dedicated wiki page. Instead, README should feature workflows and user cases. For example:
```
(all of those can be drawn as nodes in a workflow graph, with arrows, etc)
model in blender(picture) -> export to gltf ->
user loading code with camera/controls (hidden behind a cut, but easily expandable) -> 
(those merged into) -> window rendering the gltf
-> another piece of code under cut ->  animated gltf (gif? doesn't matter)
-> another piece of code under cut -> UI
...
(profit) -> screenshot of an actual game (sorta)
```

There definitely need to be pictures in README, more pictures. Not just the screenshot, but architectural ones - for users to grasp what you are building instantly. I find gfx-rs [HAL section](https://github.com/gfx-rs/gfx#hardware-abstraction-layer) to be representative.

**Comment by MrMinimal** [comment]
While we are at it, some said it would be nice to have the Discord server mentioned in the Readme.
Not just as a badge but also in the Questions / Help section like the others.

**Comment by Rhuagh** [comment]
@kvark I don't think code examples add a lot of value in a README for a game engine. I think presenting the strengths of the engine is far more important. Screenshots of actual games built on the engine would be fantastic.

File: README.md [hunk 1]
```diff
@@ -17,59 +17,33 @@
...
-[tw]: https://www.amethyst.rs/
+## Features
 
-### [Documentation][ar]
+Please visit the [features page][feat] for a list of features Amethyst provides and will provide.
 
-[ar]: https://www.amethyst.rs/doc/
+[feat]: docs/FEATURES.md
 
-## Vision
```
Comment (jojolepro) [inline README.md]: Could have kept this part, its what actually made me join the project after all ;)

Mainly those 2:
-* Massively parallel architecture.
 -* [Correct entity-component-system model][em], with entities and prefabs

File: docs/FEATURES.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Features
+
+## Existing
+
+* [x] modular structure while providing a quick solution to start your project (the main crate)
+* [x] powered by the parallel ECS library [Specs]
+* [x]  [gfx]-based rendering engine with high customizability
```
Comment (jojolepro) [inline docs/FEATURES.md]: Spacing error?

File: docs/FEATURES.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Features
```
Comment (jojolepro) [inline docs/FEATURES.md]: Listing all features in a file doesn't seem like a good idea to me. The "projects" tab on github would be a much better place to put those in, in my opinion.

Comment (jojolepro) [reply docs/FEATURES.md]: Thing is, each time we'll make a PR adding a feature we'll need to think about ticking the box in that file.

Comment (Rhuagh) [reply docs/FEATURES.md]: Projects are for future features, not existing ones imho.

Comment (MrMinimal) [reply docs/FEATURES.md]: When looking at amethyst first I had no idea what it actually did. Only after looking at examples I got it. I am for a list of the _main_ features, but no need to keep everything up to date an 100% accurate at all times.

**Comment by Xaeroxe** [comment]
@jojolepro I fixed the spacing issue.  I'd really like to get this merged personally, I view it as a big improvement over our current readme.

**Comment by torkleyy** [comment]
Thanks for all the reviews!

bors r=Rhuagh,Xaeroxe

**Pull request merged**